src/isar/storage/blob_storage.py

```diff
@@ -23,44 +26,47 @@ def __init__(
         ).value
         self.container_name = container_name
 
-        self.blob_service_client = self.get_blob_service_client()
-        self.container_client = self.get_container_client()
+        self.blob_service_client = self._get_blob_service_client()
+        self.container_client = self._get_container_client()
 
-    def store(self, data: bytes, path: Path) -> bool:
-        blob_client = self.get_blob_client(path)
+        self.logger = logging.getLogger("uploader")
+
+    def store(self, inspection: Inspection, metadata: MissionMetadata):
+        local_path, local_metadata_path = construct_local_paths(
+            inspection=inspection, metadata=metadata
+        )
+
+        metadata_bytes: bytes = construct_metadata_file(
+            inspection=inspection, metadata=metadata, filename=local_path.name
+        )
+
+        self._upload_file(path=local_path, data=inspection.data)
+        self._upload_file(path=local_metadata_path, data=metadata_bytes)
+
+    def _upload_file(self, path: Path, data: bytes):
+        blob_client = self._get_blob_client(path)
         try:
-            blob_client.upload_blob(data)
-            return True
+            blob_client.upload_blob(data=data)
         except ResourceExistsError as e:
-            logging.error(
+            self.logger.error(
                 f"Blob {path.as_posix()} already exists in container. Error: {e}"
             )
-            return False
+            raise StorageException from e
         except Exception as e:
-            logging.error(
-                f"An error occurred while uploading blob from file. Error: {e}"
-            )
-            return False
-
-    def blob_exists(self, path_to_blob: Path) -> bool:
-        blob_client = self.get_blob_client(path_to_blob)
-        try:
-            blob_client.get_blob_properties()
-            return True
-        except ResourceNotFoundError:
-            return False
+            self.logger.error(f"An unexpected error occurred while uploading blob")
+            raise StorageException from e
 
-    def get_blob_service_client(self) -> BlobServiceClient:
+    def _get_blob_service_client(self) -> BlobServiceClient:
         try:
             return BlobServiceClient.from_connection_string(
                 self.storage_connection_string
             )
         except Exception as e:
-            logging.error(f"Unable to retrieve blob service client. Error: {e}")
+            self.logger.error(f"Unable to retrieve blob service client. Error: {e}")
             raise e
 
-    def get_container_client(self) -> ContainerClient:
+    def _get_container_client(self) -> ContainerClient:
```

Comment:
If _get_container_client depends on the blob_servivice_client, should it take it as an argument or call _get_blob_service_client itself (depending if something else use the blob_service_client?)

Reply:
Implemented